rpcserver.go

```diff
@@ -3001,9 +3028,44 @@ func (r *rpcServer) QueryProof(ctx context.Context,
 		"(universeID=%x, leafKey=%x)", universeID,
 		leafKey.UniverseKey())
 
+	// Keep a record of whether the proof type was specified by the client.
+	unspecifiedArgProofType :=
+		universeID.ProofType == universe.ProofTypeUnspecified
+	if unspecifiedArgProofType {
+		// The target proof type was unspecified. Assume that the proof
+		// type is a transfer proof (more transfer proofs exist then
+		// issuance proofs).
+		universeID.ProofType = universe.ProofTypeTransfer
```

Comment:
Hmm, maybe we should just return an error in this case? Otherwise if they want issuance, then we error out?

Reply:
The caller (client calling `QueryProof`) may not know if the proof they're querying for is a transfer or issuance proof. The universe RPC proof courier is one such caller. And the caller shouldn't need to specify the proof type. Proof type specification only helps the server figure out where the proof is stored. In other words, the client already provides the necessary args for proof identification.